test/utils.jl

```diff
@@ -0,0 +1,102 @@
+const BUCKET_DATE_FORMAT = dateformat"yyyymmdd\THHMMSS\Z"
+
+is_aws(config) = config isa AWSConfig
+AWS.aws_account_number(::Minio.MinioConfig) = "123"
+
+function minio_server(body, dirs=[mktempdir()]; address="localhost:9005")
+    server = Minio.Server(dirs; address)
+
+    try
+        run(server; wait=false)
+        sleep(0.5)  # give the server just a bit of time, though it is amazingly fast to start
+
+        config = MinioConfig(
+            "http://$address"; username="minioadmin", password="minioadmin"
+        )
+        body(config)
+    finally
+        # Make sure we kill the server even if a test failed.
+        kill(server)
+    end
+end
+
+function gen_bucket_name(prefix="awss3.jl.test.")
+    # # https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+    return lowercase(prefix * Dates.format(now(Dates.UTC), BUCKET_DATE_FORMAT))
+end
+
+function assume_role(
+    aws_config::AbstractAWSConfig, role; duration=nothing, mfa_serial=nothing
+)
+    if startswith(role, "arn:aws:iam")
+        role_arn = role
+        role_name = basename(role)
+    else
+        response = AWSServices.sts(
+            "GetCallerIdentity";
+            aws_config,
+            feature_set=AWS.FeatureSet(; use_response_type=true),
+        )
+        account_id = parse(response)["GetCallerIdentityResult"]["Account"]
+        role_name = role
+        role_arn = "arn:aws:iam::$account_id:role/$role_name"
+    end
+
+    role_session = AWS._role_session_name(
+        "AWS.jl-role-",
+        role_name,
+        "-" * Dates.format(now(UTC), dateformat"yyyymmdd\THHMMSS\Z"),
+    )
+    params = Dict{String,Any}("RoleArn" => role_arn, "RoleSessionName" => role_session)
+    if duration !== nothing
+        params["DurationSeconds"] = duration
+    end
+    if mfa_serial !== nothing
+        params["SerialNumber"] = mfa_serial
+        token = Base.getpass("Enter MFA code for $mfa_serial")
+        params["TokenCode"] = read(token, String)
+        Base.shred!(token)
+    end
```

Comment:
this doesn't seem to be used directly in these tests; I'm assuming it's here so that tests can be run locally when a mfa is required, but if that's not the case then it seems like ti shoudl be removed...